sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/finishAnalysis.scala

```diff
@@ -119,3 +120,25 @@ case class ReplaceCurrentLike(catalogManager: CatalogManager) extends Rule[Logic
     }
   }
 }
+
+/**
+ * Replaces casts of special datetime strings by its date/timestamp values
+ * if the input strings are foldable.
```

Comment:
Should this be placed under constant folding? cc @cloud-fan

Reply:
Yea I think so, to avoid the side effect of rule order.

Reply:
Do you mean `ConstantFolding`? but it is an optimization rule. Could you explain me, please, why do you think that placing "correctness" (analysis) rule to an optimization rule (which can be disabled potentially w/o violating of correctness) is good idea.

Reply:
Oh I see, this is in the "finish analysis" batch, then there is no rule order problem with `ConstantFolding`. LGTM